pkg/manifests/nginx.go

```diff
@@ -151,40 +153,131 @@ func newNginxIngressControllerClusterRole(conf *config.Config, ingressConfig *Ng
 		Rules: []rbacv1.PolicyRule{
 			{
 				APIGroups: []string{""},
-				Resources: []string{"endpoints", "pods", "services", "secrets", "configmaps"},
-				Verbs:     []string{"get", "watch", "list"},
+				Resources: []string{"configmaps", "endpoints", "nodes", "pods", "secrets", "namespaces"},
+				Verbs:     []string{"list", "watch"},
+			},
+			{
+				APIGroups: []string{"coordination.k8s.io"},
+				Resources: []string{"leases"},
+				Verbs:     []string{"list", "watch"},
 			},
 			{
 				APIGroups: []string{""},
-				Resources: []string{"configmaps", "events"},
-				Verbs:     []string{"*"},
+				Resources: []string{"nodes"},
+				Verbs:     []string{"get"},
+			},
+			{
+				APIGroups: []string{""},
+				Resources: []string{"services"},
+				Verbs:     []string{"get", "list", "watch"},
 			},
 			{
 				APIGroups: []string{"networking.k8s.io"},
 				Resources: []string{"ingresses"},
 				Verbs:     []string{"get", "watch", "list"},
 			},
+			{
+				APIGroups: []string{""},
+				Resources: []string{"events"},
+				Verbs:     []string{"create", "patch"},
+			},
 			{
 				APIGroups: []string{"networking.k8s.io"},
 				Resources: []string{"ingresses/status"},
-				Verbs:     []string{"*"},
+				Verbs:     []string{"update"},
 			},
 			{
 				APIGroups: []string{"networking.k8s.io"},
 				Resources: []string{"ingressclasses"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+			{
+				APIGroups: []string{"discovery.k8s.io"},
+				Resources: []string{"endpointslices"},
 				Verbs:     []string{"list", "watch", "get"},
 			},
+		},
+	}
+}
+
+func newNginxIngressControllerRole(conf *config.Config, ingressConfig *NginxIngressConfig) *rbacv1.Role {
+	return &rbacv1.Role{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Role",
+			APIVersion: "rbac.authorization.k8s.io/v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      ingressConfig.ResourceName,
+			Labels:    addComponentLabel(GetTopLevelLabels(), "ingress-controller"),
+			Namespace: conf.NS,
+		},
+		Rules: []rbacv1.PolicyRule{
+			{
+				APIGroups: []string{""},
+				Resources: []string{"namespaces"},
+				Verbs:     []string{"get"},
+			},
+			// temporary permission used for update from 1.3.0->1.8.1
+			{
+				APIGroups: []string{""},
+				Resources: []string{"configmaps"},
+				Verbs:     []string{"update"},
+			},
+			{
+				APIGroups: []string{""},
+				Resources: []string{"configmaps", "pods", "secrets", "endpoints"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+			{
+				APIGroups: []string{""},
+				Resources: []string{"services"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
+			{
+				APIGroups: []string{"networking.k8s.io"},
+				Resources: []string{"ingresses"},
+				Verbs:     []string{"get", "list", "watch"},
+			},
 			{
 				APIGroups: []string{""},
 				Resources: []string{"nodes"},
```

Comment:
are nodes needed for the 1.3.0 transition? might be good to document that's why it's here if so.

Reply:
it matches what the nginx helm chart has 
https://github.com/kubernetes/ingress-nginx/blob/616d7e97d006cb0d054706dd0db5639497abee54/charts/ingress-nginx/templates/clusterrole.yaml#L49C1-L54C12

Reply:
that's on the clusterrole. This is on the role unless the git diff just looks extremely weird for me

Reply:
going to test if it's actually needed, as it does seem unexpected

Reply:
https://github.com/kubernetes/ingress-nginx/blob/616d7e97d006cb0d054706dd0db5639497abee54/charts/ingress-nginx/templates/controller-role.yaml 

just don't see it here which is why i'm wondering

Reply:
no, you're right, it's not on role....i misread where your comment was i think?